testsuite/tests/typing-modules/pr7726.ml

```diff
@@ -0,0 +1,139 @@
+(* TEST
+   * expect
+*)
+
+module type T = sig type t end
+module Fix(F:(T -> T)) = struct
+  module rec Fixed : T with type t = F(Fixed).t = F(Fixed)
+end;;
+[%%expect{|
+module type T = sig type t end
+module Fix :
+  functor (F : T -> T) ->
+    sig module rec Fixed : sig type t = F(Fixed).t end end
+|}]
+
+module T1 = Fix(functor (X:sig type t end) -> struct type t = X.t option end);;
+[%%expect{|
+Line _, characters 12-77:
+  module T1 = Fix(functor (X:sig type t end) -> struct type t = X.t option end);;
```

Comment:
This over-long line I think is unavoidable, so this test file needs an `ocaml-typo=long-line` entry in `.gitattributes`

Reply:
Adding `long-line` was not enough. I also had to add `missing-header`. Looks like adding an exception discards the previous ones.